### ISSUE TITLE
Sequential and parallel checkers

### ISSUE THREAD

**Issue by zenzike** [scrooloose/syntastic#744]
Currently it is possible to chain checkers together, where the following line runs one checker after the other, and stops the sequence as soon as one of the checkers fails:

```
let g:syntastic_php_checkers=['php', 'phpcs', 'phpmd']
```

A different way of executing these three checkers would be to run them all, and collect the errors together to be reported in a single sweep (I'll call this running the checkers in 'parallel').

Perhaps it would be worth having syntax such as:

```
let g:syntastic_php_checkers=[['php', 'phpcs'], 'phpmd']
```

Here things in the outer list are run in sequence, as before, but when a list of checkers is encountered, they are treated as a single checker where they are all run and errors collected. In this example, 'php' and 'phpcs' would both be run in parallel, and errors collected. If there are no errors, then 'phpmd' would be run as usual.

In other words, the outer list is a list of sequential checkers, and the inner list is in parallel.
